walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,97 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> result = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> timeRanges = new ArrayList<TimeRange>();
+    ArrayList<Integer> startTimes = new ArrayList<Integer>();
+    ArrayList<Integer> endTimes = new ArrayList<Integer>();
+
+    // If the duration is longer than the entire day, return an empty collection. 
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return result;
+    }
+    
+    // If there are no events or attendees in the request, return the whole day. 
+    if (events.isEmpty() || request.getAttendees().isEmpty()) {
+        timeRanges.add(TimeRange.WHOLE_DAY);
+        result.add(TimeRange.WHOLE_DAY);
+        return result;
+    }
+
+    // Loop through the list of events to add the start and end times for an event of the requested attendees. 
+    for (Event event : events) {
+        for (String eventPerson : event.getAttendees()) {
+            if (request.getAttendees().contains(eventPerson)) {
+                startTimes.add(event.getWhen().start());
+                endTimes.add(event.getWhen().end());
+            } 
+        }
+    }
+
+    // Check if there are any start and end times to generate meeting times, otherwise, return the whole day.
+    if (startTimes.size() > 0 && endTimes.size() > 0) {
+        // Sort the start and end times to simplify further processing. 
+        Collections.sort(startTimes);
+        Collections.sort(endTimes);
+
+        // If the first start time does not begin with 0, include a time starting from 0 to the first start time. 
+        if (startTimes.get(0) != 0) {
+            timeRanges.add(TimeRange.fromStartEnd(0, startTimes.get(0), false));
+            result.add(TimeRange.fromStartEnd(0, startTimes.get(0), false));
+        }
+
+        // If the last end time does not end with 1440, include a time ending from 1440 with the last end time. 
+        if (endTimes.get(endTimes.size()-1) != TimeRange.WHOLE_DAY.duration()) {
+            timeRanges.add(TimeRange.fromStartEnd(endTimes.get(endTimes.size()-1), 1440, false));
```

Comment:
You missed replacing 1440 with TimeRange.WHOLE_DAY, here and below. Please fix before merging.

Reply:
I've updated the values.